### PR TITLE
HTBHF-2704 Updated the random nino generator to match the latest rege…

### PR DESCRIPTION
**Pull request by dwybourn**
Updated the random nino generator to match the latest regex on the claim service.

### CODE REVIEW

File: src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala [hunk 1]
```diff
@@ -47,15 +51,25 @@ class ClaimSimulation extends Simulation {
...
   }
 
-  private def getRandomAlphabetCharAsString() = {
-    val randomChar = (random.nextInt(26) + 'A').asInstanceOf[Char]
+  private def getRandomNinoCharAsAString: String = {
+    val randomChar = VALID_NINO_CHARS.charAt(random.nextInt(VALID_NINO_CHARS.length))
     randomChar.toString
   }
 
+  private def getRandomValidTwoNinoCharsAsAString = {
```
Comment (dwybourn) [inline src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala]: I tried implementing this with recursion but that kept allowing invalid sequences through. Maybe someone with more Scala knowledge can figure out what's wrong with this code:
```private def getRandomTwoNinoCharsAsAString(): String = {
    val randomChars = getRandomNinoCharAsAString + getRandomNinoCharAsAString
    if (INVALID_NINO_FIRST_TWO_CHARS contains randomChars) {
      println randomChars
      getRandomTwoNinoCharsAsAString()
    }

    randomChars
  }```

Comment (mikeSmith08) [reply src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala]: I don't think anyone on this PR has any more Scala knowledge, so as long as it works I'm ok with it.

File: src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala [hunk 2]
```diff
@@ -47,15 +51,25 @@ class ClaimSimulation extends Simulation {
...
+    } while(INVALID_NINO_FIRST_TWO_CHARS contains randomChars)
+
+    randomChars
+  }
+
   val randomNinos = Iterator.continually(
     // Random number will be accessible in session under variable "OrderRef"
     Map("nino" -> {
-      getRandomAlphabetCharAsString + getRandomAlphabetCharAsString + ninoFormat.format(random.nextInt(999999)) + "D"
+      getRandomValidTwoNinoCharsAsAString + ninoFormat.format(random.nextInt(999999)) + "D"
```
Comment (mikeSmith08) [inline src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala]: I'm guessing we don't care what the first two chars of the NINO are for the performance tests?

File: src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala (hunk 1 above)
Comment (YetAnotherMatt) [reply src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala]: I don't think the line `getRandomTwoNinoCharsAsAString()` is returning from the method

File: src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala (hunk 2 above)
Comment (YetAnotherMatt) [reply src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala]: I don't think the performance tests check what the response contains

Comment (dwybourn) [reply src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala]: Yep, we've always been generating the nino completely randomly which will give a mix of active and rejected claims.

**Pull request merged**